Source/WebCore/page/Page.cpp

```diff
@@ -5473,7 +5473,12 @@ std::optional<TextIndicatorData> Page::textPreviewDataForActiveWritingToolsSessi
         return std::nullopt;
     }
 
-    return IntelligenceTextEffectsSupport::textPreviewDataForRange(*localTopDocument, *scope, rangeRelativeToSessionRange);
+    RefPtr textIndicator = IntelligenceTextEffectsSupport::textPreviewDataForRange(*localTopDocument, *scope, rangeRelativeToSessionRange);
+
+    if (textIndicator)
```

Comment:
usually we prefer early returns to be the "bad" case aka when textIndicator is nil, but this is fine